Documentation/contributor-guide/contributor-membership.md

```diff
@@ -0,0 +1,151 @@
+- Expected to be responsive to review requests
+- Assigned PRs to review related to area of expertise
+- Assigned test bugs related to area of expertise
+- Granted "triage access" to etcd project
+
+## Maintainers
+
+Maintainers are first and foremost contributors that have shown they
+are committed to the long term success of a project. Maintainership is about building
+trust with the current maintainers and being a person that they can
+depend on to make decisions in the best interest of the project in a consistent manner.
+
+**Defined by:** *maintainers* entry in the [MAINTAINERS] file.
+
+### Requirements
+
+- Deep understanding of the technical goals and direction of the project
+- Deep understanding of the technical domain of the project
+- Sustained contributions to design and direction by doing all of:
+    - Authoring and reviewing proposals
+    - Initiating, contributing and resolving discussions (emails, GitHub issues, meetings)
+    - Identifying subtle or complex issues in designs and implementation PRs
+- Directly contributed to the project through implementation and / or review
+- Subscribed to [etcd-maintainers@googlegroups.com]
+- Elected by supermajority of active maintainers.
+
+### Responsibilities and privileges
+
+- Make and approve technical design decisions
+- Set technical direction and priorities
+- Define milestones and releases
+- Mentor and guide reviewers, and contributors to the project.
+- Participate when called upon in the [security disclosure and release process]
+- Ensure continued health of the project
+    - Adequate test coverage to confidently release
+    - Tests are passing reliably (i.e. not flaky) and are fixed when they fail
+- Ensure a healthy process for discussion and decision making is in place.
+- Work with other maintainers to maintain the project's overall health and success holistically
+
+### Retiring
+
+Life priorities, interests, and passions can change. Maintainers can retire and
+move to [emeritus maintainers]. If a maintainer needs to step down, they should
+inform other maintainers, if possible, help find someone to pick up the related
+work. At the very least, ensure the related work can be continued. Afterward
+they can remove themselves from list of existing maintainers.
+
+If a maintainer has not been performing their duties for period of 12 months,
+they can be removed by other maintainers. In that case inactive maintainer will
+be first notified via an email. If situation doesn't improve, they will be
+removed. If an emeritus maintainer wants to regain an active role, they can do
+so by renewing their contributions. Active maintainers should welcome such a move.
+Retiring of other maintainers or regaining the status should require approval
+of at least two active maintainers.
+
+## Acknowledgements
+
+Contributor roles and responsibilities were written based on [Kubernetes community membership]
+
+[MAINTAINERS]: /MAINTAINERS
```

Comment:
No strong opinion on this, but be good to discuss it separately.

Reply:
I think that would be overkill for the project. The current structure seems good enough, but yes agree that it can be discussed separately in the future.

Reply:
In my opinion using OWNERS are great for a big project to scale reviews, however I don't think etcd project will or should grow further. We are also already splitting the codebase into separate repos which allows us to avoid per directory owners. 

I think that adding OWNERS is not aligned with project direction which is to simplify and reduce size of codebase.

Reply:
I'm seeing OWNERS as a way to give 'approvers' privileges to less sensitive parts of the code-base like tests, tools or docs that might be an additional engagement tool. 
I agree with direction to limit scope of the code-base, but even if we cut 'proxy' out of `etcd-io/etcd` repo and put it as `etcd-io/etcd-proxy`, than it might make sense (if anyone is willing to) to have separate approvers/writers for this repo (this means we would have sub-project OWNERS, but not implemented using the OWNERs file yet).